### DISCUSSION TITLE
tree-sitter-d2

### DISCUSSION THREAD

**Discussion by pleshevskiy** [terrastruct/d2#404] [Show and tell]
Thanks for such a wonderful project) I use it daily now) I thought it would be great to make a tree-sitter grammar for this project. It's not completely finished yet, but I've tried to implement, test and publish most things)

Anyone interested can try it!
https://github.com/pleshevskiy/tree-sitter-d2

For me, these grammars completely replace the need for d2-vim

**Comment by janydoe** [comment]
### Mirrors
- https://git.pleshevski.ru/pleshevskiy/tree-sitter-d2
- https://codeberg.org/p8i/tree-sitter-d2
